cmd/dcrdata/explorer/explorerroutes.go

```diff
@@ -2766,3 +2769,111 @@ func (exp *explorerUI) AttackCost(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	io.WriteString(w, str)
 }
+
+type verifyMessageResult struct {
+	Address   string
+	Signature string
+	Message   string
+	Valid     bool
+	Error     string
+}
+
+// VerifyMessagePage is the page handler for "GET /verify-message" path.
+func (exp *explorerUI) VerifyMessagePage(w http.ResponseWriter, r *http.Request) {
+	str, err := exp.templates.exec("verify_message", struct {
+		*CommonPageData
+		VerifyMessageResult *verifyMessageResult
+	}{
+		CommonPageData: exp.commonData(r),
+	})
+	if err != nil {
+		log.Errorf("Template execute failure: %v", err)
+		exp.StatusPage(w, defaultErrorCode, defaultErrorMessage, "", ExpStatusError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/html")
+	w.WriteHeader(http.StatusOK)
+	io.WriteString(w, str)
+}
+
+// VerifyMessageHandler is the handler for "POST /verify-message" path.
+func (exp *explorerUI) VerifyMessageHandler(w http.ResponseWriter, r *http.Request) {
+	address := r.PostFormValue("address")
+	signature := r.PostFormValue("signature")
+	message := r.PostFormValue("message")
+
+	displayPage := func(msg string, result bool) {
+		str, err := exp.templates.exec("verify_message", struct {
+			*CommonPageData
+			VerifyMessageResult *verifyMessageResult
+		}{
+			CommonPageData: exp.commonData(r),
+			VerifyMessageResult: &verifyMessageResult{
+				Address:   address,
+				Signature: signature,
+				Message:   message,
+				Valid:     result,
+				Error:     msg,
+			},
+		})
+
+		if err != nil {
+			log.Errorf("Template execute failure: %v", err)
+			exp.StatusPage(w, defaultErrorCode, defaultErrorMessage, "", ExpStatusError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "text/html")
+		w.WriteHeader(http.StatusOK)
+		io.WriteString(w, str)
+	}
+
+	if address == "" || signature == "" || message == "" {
+		displayPage("Form values cannot be empty", false)
+		return
+	}
+
+	// Decode the provided address.  This also ensures the network encoded with
+	// the address matches the network the server is currently on.
+	addr, err := stdaddr.DecodeAddress(address, exp.ChainParams)
```

Comment:
Rather than repeating all of this logic, why not just use https://pkg.go.dev/github.com/decred/dcrd/dcrutil/v4#VerifyMessage?

```Go
if err := dcrutil.VerifyMessage(address, signature, message, exp.ChainParams); err != nil {
	/* displayPage(....) */
	return
}
```

Reply:
Ah good point.  I had only come across dcrwallet's version before so I suggested starting from there.  Definitely use the exported dcrutil function though!

Reply:
@davecgh @chappjc thanks for the review.